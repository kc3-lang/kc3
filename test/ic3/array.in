(U8) {0, 1}
a = (U8) {0, 1}
quote a[0]
quote a[1]
a[0]
a[1]
(U8) {{0, 1},
      {2, 3}}
b = (U8) {{0, 1},
          {2, 3}}
quote b[0][0]
quote b[0][1]
quote b[1][0]
quote b[1][1]
b[0][0]
b[0][1]
b[1][0]
b[1][1]
c = (U8) {{{0, 1},
           {2, 3}},
          {{4, 5},
           {6, 7}}}
quote c[0][0][0]
quote c[0][0][1]
quote c[0][1][0]
quote c[0][1][1]
quote c[1][0][0]
quote c[1][0][1]
quote c[1][1][0]
quote c[1][1][1]
c[0][0][0]
c[0][0][1]
c[0][1][0]
c[0][1][1]
c[1][0][0]
c[1][0][1]
c[1][1][0]
c[1][1][1]
d = (U8) {{{{0, 1},
            {2, 3}},
           {{4, 5},
            {6, 7}}},
          {{{8, 9},
            {10, 11}},
           {{12, 13},
            {14, 15}}}}
quote d[0][0][0][0]
quote d[0][0][0][1]
quote d[0][0][1][0]
quote d[0][0][1][1]
quote d[0][1][0][0]
quote d[0][1][0][1]
quote d[0][1][1][0]
quote d[0][1][1][1]
quote d[1][0][0][0]
quote d[1][0][0][1]
quote d[1][0][1][0]
quote d[1][0][1][1]
quote d[1][1][0][0]
quote d[1][1][0][1]
quote d[1][1][1][0]
quote d[1][1][1][1]
d[0][0][0][0]
d[0][0][0][1]
d[0][0][1][0]
d[0][0][1][1]
d[0][1][0][0]
d[0][1][0][1]
d[0][1][1][0]
d[0][1][1][1]
d[1][0][0][0]
d[1][0][0][1]
d[1][0][1][0]
d[1][0][1][1]
d[1][1][0][0]
d[1][1][0][1]
d[1][1][1][0]
d[1][1][1][1]
e = (U8) { 1 + 1 }
e[0]
f = (U8) { 1 + 1, 2 + 2 }
f[0]
f[1]
g = (U8) { 1 + 1, 2 + 2, 3 + 3 }
g[0]
g[1]
g[2]
h = (U8) { 255 + 1, 255 + 2, 255 + 3 }
h[0]
h[1]
h[2]
i = (U16) { 255 + 1, 255 + 2, 255 + 3 }
i[0]
i[1]
i[2]
j = (U32) { 255 + 1, 255 + 2, 255 + 3 }
j[0]
j[1]
j[2]
k = (U64) { 255 + 1, 255 + 2, 255 + 3 }
k[0]
k[1]
k[2]
quote (Integer) { 1000000000000000000000000000000001,
                  2000000000000000000000000000000002 }
l = (Integer) { 1000000000000000000000000000000001,
                2000000000000000000000000000000002 }
l[0]
l[1]
quote (List) {[1, 2], [3, 4]}
(List) {[1, 2], [3, 4]}
m = (List) {[1, 2], [3, 4]}
m[0]
m[1]
