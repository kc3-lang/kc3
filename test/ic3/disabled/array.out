(U8) {0, 1}
(U8) {0, 1}
a[0]
a[1]
0
1
(U8) {{0, 1}, {2, 3}}
(U8) {{0, 1}, {2, 3}}
b[0][0]
b[0][1]
b[1][0]
b[1][1]
0
1
2
3
(U8) {{{0, 1}, {2, 3}}, {{4, 5}, {6, 7}}}
